### PR TITLE
Add workspace status script

### PR DESCRIPTION
**Pull request by pudelkoM**
The workspace script collects information that can be used to stamp binaries and other Bazel artifacts in the future.
It extends the existing files `./bazel-out/stable-status.txt` and `./bazel-out/volatile-status.txt`.

```
BUILD_TIMESTAMP 1610129464
GIT_REF b56022e7692ecaca94c2f6cc69761fd9145fa6e0-dirty
GIT_SHA b56022e7692ecaca94c2f6cc69761fd9145fa6e0-dirty
GIT_URL https://github.com/stratum/stratum.git
```

Use cases could be PR #489 or issue #475

### CODE REVIEW

File: .circleci/workspace_status.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# If the script exits with non-zero code, it's considered as a failure
+# and the output will be discarded.
+
+# The code below presents an implementation that works for git repository
+git_rev=$(git rev-parse HEAD)
+if [[ $? != 0 ]];
+then
+    exit 1
+fi
+echo "BUILD_SCM_REVISION ${git_rev}"
```
Comment (pudelkoM) [inline .circleci/workspace_status.sh:22]: Exact values tbd

**Comment by Yi-Tseng** [comment]
can we use `set -e -o pipefail` so we don't have to check the result from every output?

**Comment by pudelkoM** [comment]
> can we use `set -e -o pipefail` so we don't have to check the result from every output?

Definitely worth trying. The script is still a little WIP.

**Review by bocon13 — APPROVED** [review]
looks reasonable

**Pull request merged**